pypeit/edgetrace.py

```diff
@@ -394,92 +371,141 @@ class EdgeTraceSet(object):
+        if not isinstance(traceimg, TraceImage):
+            msgs.error('Input trace_mg must be a TraceImage object.')
         if not isinstance(spectrograph, Spectrograph):
             msgs.error('Input spectrograph must be a Spectrograph object.')
         if not isinstance(par, EdgeTracePar):
             msgs.error('Input par must be an EdgeTracePar object.')
 
-        self.trace_img = trace_img      # Input  TraceImage
-
-        # TODO: just use trace_img for many of these attributes
-        self.img = trace_img.image      # The image used to find the slit edges
-        self.binning = self.trace_img.detector.binning  # Detector ordered:  spec, spat
-        self.det = self.trace_img.detector.det          # Detector used for the trace image
-        self.nspec, self.nspat = self.img.shape         # The shape of the trace image
-        self.files = self.trace_img.files               # Files used to construct the trace image
-
-        self.spectrograph = spectrograph    # Spectrograph used to take the data
-        self.PYP_SPEC = spectrograph.spectrograph  # For the Header.  Will be in datamodel
-        self.par = par                      # Parameters used for slit edge tracing
-
-        self.bpm = None                 # Mask for the trace image
-        self.sobel_sig = None           # Sobel filtered image used to detect edges
-        self.sobel_sig_left = None      # Sobel filtered image used to trace left edges
-        self.sobel_sig_right = None     # Sobel filtered image used to trace right edges
-        # TODO: Need a separate mask for the sobel image?
-
-        self.traceid = None             # The ID numbers for each trace
-        self.orderid = None             # For echelle spectrographs, this is the matched order ID
-                                        # of each trace. Left traces are negative, right traces
-                                        # are positive, and the absolute value of both is the
-                                        # echelle order.  See match_order().
-
-        self.spat_img = None            # (Integer) Pixel nearest the slit edge for each trace
-        self.spat_cen = None            # (Floating-point) Spatial coordinate of the slit edges
-                                        # for each spectral pixel
-        self.spat_err = None            # Error in the slit edge spatial coordinate
-        self.spat_msk = None            # Mask for the slit edge position for each spectral pixel
-
-        self.spat_fit = None            # The result of modeling the slit edge positions
-        self.spat_fit_type = None       # The type of fitting performed
-        
-        self.pca = None                 # One or two TracePCA objects with PCA decomposition
-        self.pca_type = None            # Measurements used to construct the PCA (center or fit)
-
+        # TODO:
+        #   - Change spectrograph.spectrograph to spectrograph.name
```

Comment:
I support this

Reply:
Leaving the TODO, but punting in this PR.